### PR TITLE
F/copyright review 2015

### PR DESCRIPTION
**Pull request by bkil-syslogng**
EDIT: If you agree to retrospectively adjust the old copyright headers as well, you can close PR #822 and merge the extended version instead: https://github.com/balabit/syslog-ng/pull/823

PR #822 does not contain the following proposed automatic modifications.
(I would also do merging around s.a.r.l lines manually,
because the legal entity did not change and lived in continuity.)

```bash
find . -type f |
while read FILE; do
sed --silent --regexp-extended "
 s~\<Portions created by ~&~
 t skip
 s~^ \* Copyright \(c\) 1990, 1993$~&~
 t skip

s~^((.*)\<[cC]opyright \([cC]\) ([0-9,. -]+[0-9])\
( Bala[bB]it( IT Ltd(, Budapest, Hungary)?\
| S\.a\.r\.l\., Luxembourg, Luxembourg)?)?)$~\
-\1\n\
+\2Copyright (c) \3~

 T skip
 s~2015$~&~
 t ok
 s~-2014 *$~-2015~
 t ok
 s~(2014) *$~\1-2015~
 t ok
 s~$~, 2015~
 :ok
 s~$~ Balabit~
 p
 :skip
" \
"$FILE" || break
done |
grep --line-buffered --extended-regexp --color=always "(Balabit)?$" |
less --RAW-CONTROL-CHARS
```

### CODE REVIEW

File: lib/file-perms.c [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
  * Copyright (c) 2012 BalaBit IT Ltd, Budapest, Hungary
```
Comment (bazsi) [inline lib/file-perms.c]: Balabit it and balabit sarl should also be converted to Balabit, without entity form or location.

Comment (bazsi) [inline lib/file-perms.c]: Otherwise looks good to me,and even the rest can be a separate pr.

Comment (bkil-syslogng) [inline lib/file-perms.c]: @bazsi I didn't bother to change it at this location, because the specified script will automatically correct all occurrences, including this one. Actually I did bother and corrected it by hand, but I stashed all manual corrections and decided to express it in sed. If you support such an upgrade, I'll submit the result of that script in a different PR.

Comment (bazsi) [inline lib/file-perms.c]: I thought the patch is the result of the script. When would those changes
be applied?
On Dec 2, 2015 5:56 PM, "Tamas Nagy" <notifications@github.com> wrote:

In lib/file-perms.c
<https://github.com/balabit/syslog-ng/pull/822#discussion_r46440627>:

> @@ -1,6 +1,6 @@
>  /*
>   * Copyright (c) 2012 BalaBit IT Ltd, Budapest, Hungary

@bazsi <https://github.com/bazsi> I didn't bother to change it at this
location, because the specified script will automatically correct all
occurrences, including this one. Actually I did bother and corrected it by
hand, but I stashed all manual corrections and decided to express it in
sed. If you support such an upgrade, I'll submit the result of that script
in a different PR.

—
Reply to this email directly or view it on GitHub
<https://github.com/balabit/syslog-ng/pull/822/files#r46440627>.

Comment (bkil-syslogng) [inline lib/file-perms.c]: @bazsi I've submitted the whole thing as a separate PR. I include the script used for transformation when a commit is not done manually.
https://github.com/balabit/syslog-ng/pull/823

Comment (bazsi) [inline lib/file-perms.c]: OK, great then. The company code looks great to me.

I had one 2nd though about the java bindings. It's clear that we want the
plugins under the gpl, but the library part we might want to license that
under the lgpl to permit java plugins that would depend in gpl incompatible
libs.

In lib/file-perms.c
<https://github.com/balabit/syslog-ng/pull/822#discussion_r46446643>:

> @@ -1,6 +1,6 @@
>  /*
>   * Copyright (c) 2012 BalaBit IT Ltd, Budapest, Hungary

@bazsi <https://github.com/bazsi> I've submitted the whole thing as a
separate PR. I include the script used for transformation when a commit is
not done manually.
#823 <https://github.com/balabit/syslog-ng/pull/823>

—
Reply to this email directly or view it on GitHub
<https://github.com/balabit/syslog-ng/pull/822/files#r46446643>.

Comment (bkil-syslogng) [inline lib/file-perms.c]: You make a valid point and we're just at the right time to clear up such licensing questions. All input is welcome.

**Comment by lbudai** [comment]
All the patches inside this PR are contained by #823 .
After talked with @bkil-syslogng , I'll close this.
